### PR TITLE
Expose new API for image component

### PR DESCRIPTION
**Pull request by baseplate-admin**

### CODE REVIEW

File: src/routes/[...lang=locale]/anime/+page.svelte [hunk 1]
```diff
@@ -139,43 +139,43 @@
...
+			timestamp: "3 hour ago" // TODO: format time
 		}
 	];
 	// state for latest_episodes color palette
 	const latest_episodes_color_palette_mapping: {
 		[key: number]: [number, number, number][];
 	} = $state({});
+	const latest_episodes_loaded_mapping: boolean[] = $state(
+		new Array(latest_episodes.length).fill(false)
+	);
```
Comment (baseplate-admin) [inline src/routes/[...lang=locale]/anime/+page.svelte:175]: This 2  needs to be merged, or we should have a better structure than this shit

File: src/routes/[...lang=locale]/anime/+page.svelte [hunk 2]
```diff
@@ -329,51 +329,65 @@
...
-					class="snap-y grid w-full grid-cols-2 grid-rows-auto auto-rows-min md:gap-[1.25vw] scroll-smooth overflow-y-scroll md:pr-[1.5vw] [scrollbar-color:rgba(255,255,255,0.12)transparent]"
+					class="grid-rows-auto grid w-full snap-y auto-rows-min grid-cols-2 overflow-y-scroll scroll-smooth [scrollbar-color:rgba(255,255,255,0.12)transparent] md:gap-[1.25vw] md:pr-[1.5vw]"
 					class:scrollbar-none={IS_CHROMIUM}
 				>
 					{#each latest_episodes as episode}
-						{@const has_color_palette = latest_episodes_color_palette_mapping[episode.id] !== undefined}
-						{@const color_palette = has_color_palette && rgbHex(...latest_episodes_color_palette_mapping[episode.id][0])}
+						{@const color_palette =
+							latest_episodes_color_palette_mapping[episode.id] !== undefined &&
+							rgbHex(...latest_episodes_color_palette_mapping[episode.id][0])}
```
Comment (baseplate-admin) [inline src/routes/[...lang=locale]/anime/+page.svelte:342]: Possibly we can remove this `undefined` check if we move to a better structure

File: src/routes/[...lang=locale]/anime/+page.svelte [hunk 3]
```diff
@@ -329,51 +329,65 @@
...
-						<div
-							class="snap-start w-full md:h-[5vw] bg-cover bg-center relative md:rounded-[0.75vw] border md:border-[0.15vw] border-accent/50 duration-300"
-							style="
+						{#if image_loaded}
+							<div
+								class="relative w-full snap-start border border-accent/50 bg-cover bg-center duration-300 md:h-[5vw] md:rounded-[0.75vw] md:border-[0.15vw]"
+								style="
 								background-image: url({episode.banner});
 								border-color: #{color_palette};
 							"
```
Comment (baseplate-admin) [inline src/routes/[...lang=locale]/anime/+page.svelte:351]: Avoid using direct style and use `var()` with tailwind here

File: src/routes/[...lang=locale]/anime/+page.svelte [hunk 4]
```diff
@@ -329,51 +329,65 @@
...
+												>Ep {episode.ep_number.toString().padStart(2, "0")}</span
+											>
+											<Circle class="opacity-75 md:size-[0.25vw]" />
+											<span class="line-clamp-1">{episode.timestamp}</span>
+										</div>
 									</div>
+									<a
+										href="anime/mal/{episode.id}/episode/{episode.ep_number}"
+										class="btn h-max min-h-max rounded-full border-none bg-warning md:mr-[0.5vw] md:p-[0.75vw]"
+										style="background-color: #{color_palette};"
```
Comment (baseplate-admin) [inline src/routes/[...lang=locale]/anime/+page.svelte:377]: Avoid using direct style and use var() with tailwind here

File: src/routes/[...lang=locale]/anime/+page.svelte [hunk 5]
```diff
@@ -329,51 +329,65 @@
...
-									href="anime/mal/{episode.id}/episode/{episode.ep_number}"
-									class="rounded-full btn border-none min-h-max h-max md:p-[0.75vw] md:mr-[0.5vw] bg-warning"
-									style="background-color: #{color_palette};"
-								>
-									<Play class="md:size-[1.25vw]" />
-								</a>
 							</div>
-						</div>
+						{:else}
+							<!-- show skeleton -->
```
Comment (baseplate-admin) [inline src/routes/[...lang=locale]/anime/+page.svelte:384]: Skeleton here

**Review by baseplate-admin — COMMENTED** [review]
Overall good approach but needs to be cleaned up

File: src/routes/[...lang=locale]/anime/+page.svelte (hunk 5 above)
Comment (tokitouq) [reply src/routes/[...lang=locale]/anime/+page.svelte]: Resolved on https://github.com/baseplate-admin/coreproject-v4-ui/pull/77/commits/6099f93634b072dc8a54151c52b326669820ea56

File: src/routes/[...lang=locale]/anime/+page.svelte (hunk 4 above)
Comment (tokitouq) [reply src/routes/[...lang=locale]/anime/+page.svelte:380]: Resolved on https://github.com/baseplate-admin/coreproject-v4-ui/pull/77/commits/58859ef6cf58e42a95267d9bad6f91b93a3234ee

File: src/routes/[...lang=locale]/anime/+page.svelte (hunk 3 above)
Comment (tokitouq) [reply src/routes/[...lang=locale]/anime/+page.svelte]: Resolved on https://github.com/baseplate-admin/coreproject-v4-ui/pull/77/commits/58859ef6cf58e42a95267d9bad6f91b93a3234ee

File: src/routes/[...lang=locale]/anime/+page.svelte [hunk 6]
```diff
@@ -170,6 +170,14 @@
...
 		[key: number]: [number, number, number][];
 	} = $state({});
+	const latest_episodes_loaded_mapping: boolean[] = $state(
+		new Array(latest_episodes.length).fill(false)
+	);
+
+	const some_mapping = $state({
+		color_palette: {},
+		image_loaded: new Array(latest_episodes.length).fill(false)
+	});
```
Comment (baseplate-admin) [inline src/routes/[...lang=locale]/anime/+page.svelte:180]: I am still -1 on this

File: src/routes/[...lang=locale]/anime/+page.svelte [hunk 7]
```diff
@@ -334,52 +342,62 @@
...
+									<div class="rounded-full bg-neutral md:h-[1vw]"></div>
+									<div class="w-1/2 rounded-full bg-neutral md:h-[0.5vw]"></div>
+								</div>
+								<div class="rounded-full bg-neutral md:mr-[0.5vw] md:size-[2.75vw]"></div>
+							</div>
+						{/if}
 						<!-- use Image component for just to get color -->
 						<Image
 							src={episode.cover}
 							class="hidden"
```
Comment (baseplate-admin) [inline src/routes/[...lang=locale]/anime/+page.svelte:399]: This should not be `hidden`

File: src/routes/[...lang=locale]/anime/+page.svelte [hunk 8]
```diff
@@ -334,52 +342,62 @@
...
+								class="flex items-center bg-neutral/25 md:h-[5vw] md:gap-[1vw] md:rounded-[0.75vw] md:p-[0.5vw]"
+							>
+								<div class="h-full bg-neutral md:w-[2.5vw] md:rounded-[0.5vw]"></div>
+								<div class="flex flex-1 flex-col md:gap-[0.5vw]">
+									<div class="rounded-full bg-neutral md:h-[1vw]"></div>
+									<div class="w-1/2 rounded-full bg-neutral md:h-[0.5vw]"></div>
+								</div>
+								<div class="rounded-full bg-neutral md:mr-[0.5vw] md:size-[2.75vw]"></div>
+							</div>
+						{/if}
```
Comment (baseplate-admin) [inline src/routes/[...lang=locale]/anime/+page.svelte:395]: We want transition here

File: src/routes/[...lang=locale]/anime/+page.svelte (hunk 6 above)
Comment (tokitouq) [reply src/routes/[...lang=locale]/anime/+page.svelte:180]: OHH that was committed by mistake XD

File: src/routes/[...lang=locale]/anime/+page.svelte (hunk 8 above)
Comment (tokitouq) [reply src/routes/[...lang=locale]/anime/+page.svelte:395]: It has transition, take a close look

Comment (baseplate-admin) [reply src/routes/[...lang=locale]/anime/+page.svelte:395]: use `blur`

Comment (baseplate-admin) [reply src/routes/[...lang=locale]/anime/+page.svelte:395]:
```svelte
{:else}
							<div
								class="flex items-center bg-neutral/25 md:h-[5vw] md:gap-[1vw] md:rounded-[0.75vw] md:p-[0.5vw]"
							>
								<div class="h-full bg-neutral md:w-[2.5vw] md:rounded-[0.5vw]"></div>
								<div class="flex flex-1 flex-col md:gap-[0.5vw]">
									<div class="rounded-full bg-neutral md:h-[1vw]"></div>
									<div class="w-1/2 rounded-full bg-neutral md:h-[0.5vw]"></div>
								</div>
								<div class="rounded-full bg-neutral md:mr-[0.5vw] md:size-[2.75vw]"></div>
							</div>
						{/if}
						```
On this

File: src/lib/components/image/index.svelte [hunk 9]
```diff
@@ -13,7 +13,7 @@
 		src: string;
 		class?: string;
 		color_palette: number[][];
-		image_loaded?: boolean;
+		image_loaded: boolean;
```
Comment (baseplate-admin) [inline src/lib/components/image/index.svelte:16]: Please dont change this

Comment (tokitouq) [reply src/lib/components/image/index.svelte:16]: Huh? why? its causing type issues outside component

File: src/routes/[...lang=locale]/anime/+page.svelte (hunk 8 above)
Comment (tokitouq) [reply src/routes/[...lang=locale]/anime/+page.svelte]: > Use a grid on this

Even tho, output is same...
and this is unnecessary in this case

File: src/routes/[...lang=locale]/anime/+page.svelte (hunk 7 above)
Comment (baseplate-admin) [reply src/routes/[...lang=locale]/anime/+page.svelte:402]: >  cause you see everything is wrapped under {#if image_loaded}. this Image component wont work if we move this under this condition

Maybe the solution is then an aboslute-relative type thing?

File: src/lib/components/image/index.svelte (hunk 9 above)
Comment (tokitouq) [reply src/lib/components/image/index.svelte]: Resolved on https://github.com/baseplate-admin/coreproject-v4-ui/pull/77/commits/ec290b7411116a41cdcfe711b7e73b7f0f84c5be

File: src/routes/[...lang=locale]/anime/+page.svelte (hunk 1 above)
Comment (tokitouq) [reply src/routes/[...lang=locale]/anime/+page.svelte]: Resolved on https://github.com/baseplate-admin/coreproject-v4-ui/pull/77/commits/88f5310849a03bf81c48b69abd12e50153f6c5fb

**Comment by tokitouq** [comment]
Im merging this cause I need this for other coming PRs
also goal of the PR seems achieved.. :tada:

**Pull request merged**